### PR TITLE
UX: set focus on input field of modal dialog "New Folder"

### PR DESCRIPTION
**Pull request by elizavetaRa**
## Description
Modal dialog "oc-modal" gets focus on it's input field of class "oc-text-input" on open so the user can immediately start writing and using "enter" to submit.

## Motivation and Context
UX improvement: removing an unnecessary mouse click

## How Has This Been Tested?
- test: click on "New folder", start typing, press "Enter", caused expected behavior
- tested in Chrome Version 88.0.4324.182 (64-Bit)
- tested in Firefox 78.7.0esr (64-Bit)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

### CODE REVIEW

**Comment by kulmann** [comment]
Hey @elizavetaRa, thanks for your contribution! While we're super happy it, I have to say that unfortunately it's not "that easy". We have a strong requirement to solve the focus management in an [WCAG 2.1](https://www.w3.org/TR/WCAG21/) compliant way.

TL;DR: if you set the initial focus to the text input, it is not necessarily clear to  people using a screen reader what kind of entity/action they are providing input for, because the screen reader will skip the title of the modal.

Further discussion in:
- https://github.com/owncloud/web/issues/3684
- https://github.com/owncloud/web/issues/3892#issuecomment-694918182

For being able to set the focus to the input upon opening the modal, we'd need to provide a proper description for the input element, so that it's not distort UX for screen readers.

I hope this doesn't scare you away. Would you be interested in investing more time and thoughts in this?

**Comment by labkode** [comment]
@kulmann ,  may I propose the following?

The PR made by @elizavetaRa  solves a usability (not accessibility) problem for a big majority of users, certainly not all, but is a reasonably good start, specially for customers  moving from OC 10 (where the autofocus is present) to OCIS.  Autofocus is a primal usability attribute for the modern web.

Further accessibility improvements can be added at a later  phase (this PR does not exclude that). The linked issue has not been updated for the last 6 months.  

Don't wait for perfection, ship your product early on and continue improving it along the way.

**Comment by elizavetaRa** [comment]
@kulmann As the modal is reaction of the system on clicking "New"-> "New folder", skipping the heading by screen readers wouldn't have negative impact if the input field is ARIA-annotated (which is the case). I would suggest to change "New folder" to "Create new folder" similar to other action namings "Upload file", "Upload folder" by clicking "New".

**Comment by labkode** [comment]
@kulmann Can we get this in?

**Comment by kulmann** [comment]
Since accessibility requirements in the OcModal component are now fulfilled we were able to implement the autofocus directly in the Design System. See [this PR](https://github.com/owncloud/owncloud-design-system/pull/1453). Will be part of the next release (approx. 9th of July).
